lib/ansible/modules/cloud/amazon/data_pipeline.py

```diff
@@ -0,0 +1,594 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: data_pipeline
+version_added: "2.4"
+author: Raghu Udiyar <raghusiddarth@gmail.com> (@raags)
+requirements: [ "boto3" ]
+short_description: Create and manage AWS Datapipelines
+extends_documentation_fragment:
+    - aws
+    - ec2
+description:
+    - Create and manage AWS Datapipelines. Creation is not idempotent in AWS,
+      so the uniqueId is created by hashing the options (minus objects) given to the datapipeline.
+
+      The pipeline definition must be in the format given here
+      U(http://docs.aws.amazon.com/datapipeline/latest/APIReference/API_PutPipelineDefinition.html#API_PutPipelineDefinition_RequestSyntax)
+
+      Also operations will wait for a configurable amount
+      of time to ensure the pipeline is in the requested state.
+options:
+  name:
+    description:
+      - Name of the Datapipeline
+    required: true
+  description:
+    description:
+      - Optional pipeline description
+    required: false
+    default: ''
+  objects:
+    description:
+      - List of pipeline Objects
```

Comment:
Is there a link specific to the structure of these objects? I like to include as much info about complex types in the docs as possible.